spryker/usr/local/share/spryker/spryker_functions.sh

```diff
@@ -38,73 +46,126 @@ do_spryker_app_permissions() {
+  set -x
 }
 
 do_spryker_build() {
   do_spryker_directory_create
   do_spryker_config_create
+
+  if [ "$IS_APP_MOUNTPOINT" == 'true' ] || [ "${TASK}" == "build" ]; then
+    if spryker_service_zed; then
+      do_spryker_generate_files
+    fi
+    do_spryker_build_assets
+    do_spryker_app_permissions
+  fi
 }
 
-do_build_assets() {
-  # use Spryker scripts to install static assets
-  TERM=linux
-  export TERM
-  source /app/deploy/setup/util/print.sh
-  source /app/deploy/setup/frontend/params.sh
-  source /app/deploy/setup/frontend/functions.sh
-  setupYvesFrontend
-  setupZedFrontend
+spryker_build_assets() {
+  local -r TYPE="$1"
+  as_code_owner "
+    # use Spryker scripts to install static assets
+    TERM=linux
+    export TERM
+    source /app/deploy/setup/util/print.sh
+    source /app/deploy/setup/frontend/params.sh
+    source /app/deploy/setup/frontend/functions.sh
+    setup${TYPE}Frontend
+  "
+}
+
+do_spryker_build_assets() {
+  set +e
+  ! spryker_service_yves || spryker_build_assets 'Yves'
+  ! spryker_service_zed || spryker_build_assets 'Zed'
+  set -e
 }
 
-do_generate_files() {
+do_spryker_generate_files() {
   as_code_owner "vendor/bin/console setup:deploy:prepare-propel"
   as_code_owner "vendor/bin/console transfer:generate"
   as_code_owner "vendor/bin/console setup:search:index-map"
   as_code_owner "vendor/bin/console application:build-navigation-cache"
 }
 
-do_setup() {
-  do_spryker_install
-}
-
 do_spryker_install() {
   # check if database exists (it is supposed to be created by postgres container)
   set +e
     psql -U "$DATABASE_USER" -h "$DATABASE_HOST" -lqt | cut -d \| -f 1 | grep -q "$DATABASE_NAME"
     DATABASE_EXISTS=$?
   set -e
 
-  if [ "$DATABASE_EXISTS" -eq 0 ]; then
-    # database exists
-    # check if spryker is installed
-    set +e
-      psql -U spryker_user -h postgres spryker -c "SELECT EXISTS (SELECT * FROM   information_schema.tables WHERE table_catalog = '$DATABASE_NAME' AND table_name = 'spy_locale');" | grep -q f
-      SPRYKER_INSTALLED=$?
-    set -e
-
-    if [ "$SPRYKER_INSTALLED" -ne 1 ]; then
-      as_code_owner "vendor/bin/console setup:install"
-      do_import_demodata
-      do_run_collectors
-      do_setup_search
-    fi
-  else
+  if [ "$DATABASE_EXISTS" -ne 0 ]; then
     echo "Database does not exist"
+    exit 1
+  fi
+
+  # database exists
+  # check if spryker is installed
+  set +e
+    psql -U "$DATABASE_USER" -h "$DATABASE_HOST" -c "SELECT EXISTS (SELECT * FROM   information_schema.tables WHERE table_catalog = '$DATABASE_NAME' AND table_name = 'spy_locale');" | grep -q f
+    SPRYKER_INSTALLED=$?
+  set -e
+
+  if [ "$SPRYKER_INSTALLED" -ne 1 ]; then
+    as_code_owner "vendor/bin/console setup:install"
+    do_spryker_import_demodata
+    do_spryker_product_label_relations_update
+    do_spryker_run_collectors
+    do_spryker_setup_search
+  fi
+}
+
+do_spryker_migrate() {
+  do_spryker_propel_install
+
+  if [ "$APPLICATION_ENV" != "production" ]; then
```

Comment:
A danger here in that if production is only used in production then this process is only ever skipped in production and never tested skipped before production